lib/pat.c

```diff
@@ -3329,7 +3339,11 @@ grn_pat_fuzzy_search(grn_ctx *ctx,
     id = grn_pat_fuzzy_search_find_prefixed_start_node_id(ctx, &data);
   } else {
     PAT_AT(pat, GRN_ID_NIL, node);
-    id = node->lr[1];
+    if (node) {
+      id = pat_node_get_right(pat, node);
+    } else {
+      return ctx->rc;
+    }
```

Comment:
```suggestion
    if (!node) {
      return ctx->rc;
    }
    id = pat_node_get_right(pat, node);
```

チェックして `ctx->rc` をreturnするときってこういう感じ↓で書くのが多いので、ここでもそうするのはどうでしょうか。

こういう感じ: https://github.com/groonga/groonga/blob/a865753b1172860a648d624ea80dab9a37d7f000/lib/db.c#L703-L715